workflows/snvphylnfc.nf

```diff
@@ -206,6 +215,31 @@ workflow SNVPHYL {
     CUSTOM_DUMPSOFTWAREVERSIONS (
         ch_versions.unique().collectFile(name: 'collated_versions.yml')
     )
+
+}
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    SELECT REFERENCE
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+def select_reference(refgenome, reference_sample_id, sample_assemblies) {
+
+    if(refgenome) {
+        reference_genome = Channel.value(refgenome)
```

Comment:
Is there a reason why you are using `Channel.value` instead of `Channel.fromPath` here?

Reply:
Also, could you maybe have a `log.debug "Selecting reference genome ${reference_genome} from '--refgenome'"` here?

Reply:
> Is there a reason why you are using `Channel.value` instead of `Channel.fromPath` here?

So, as far as I understand from testing this and reference the docs (https://javadoc.io/doc/io.nextflow/nextflow/19.08.0-edge/nextflow/Channel.html), `Channel.value` returns a value channel (which can be read many times), whereas `Channel.fromPath` returns a queue channel (where each item is read only once -- so only one sample is run). However, I probably should use `Channel.value(file(refgenome))`.

Reply:
> Also, could you maybe have a `log.debug "Selecting reference genome ${reference_genome} from '--refgenome'"` here?

Added in 1da07b49d29bf659c2091e6ffe2dce4028688962